### PR TITLE
Remove filterfalse and Python 2 related things from compatibility

### PR DESCRIPTION
**Pull request by namannimmo10**
#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/core/compatibility.py [hunk 1]
```diff
@@ -361,10 +360,8 @@ def is_sequence(i, include=None):
             bool(include) and
             isinstance(i, include))
 
-try:
-    from itertools import zip_longest
-except ImportError:  # Python 2.7
-    from itertools import izip_longest as zip_longest
+
+from itertools import zip_longest
```
Comment (asmeurer) [inline sympy/core/compatibility.py]: This should be removed, and the imports that use it changed to import it directly from itertools (same with all other Python 2 things in this file).

Comment (namannimmo10) [reply sympy/core/compatibility.py]: Can you please rephrase the sentence?  What I can understand from your statement is that the file using `zip_longest` for instance, should import that directly from `itertools` but not from `sympy.core.compatibility`.

**Comment by namannimmo10** [comment]
@asmeurer If any changes are still required, please suggest

**Comment by asmeurer** [comment]
If you want to remove other things from compatibility then there is more that can be done here. But if you don't want to do that this can be merged as is (assuming the tests pass).

**Comment by namannimmo10** [comment]
@asmeurer I'll try to get more comfortable with the differences between Python 2 and Python 3 and maybe fix other tasks in #17943 soon. I think it's good to merge as of now :) And thanks for reviewing this PR.

**Comment by asmeurer** [comment]
OK. If you're going to clean up compatibility. I would suggest removing one function at a time (one commit per each one, which can go in one large PR).

**Comment by namannimmo10** [comment]
@asmeurer I'm trying to remove [this](https://github.com/sympy/sympy/blob/e696344ae3a8308399fe43d790880104dfe0e0b5/sympy/core/compatibility.py#L114) whole part. But how can I test on my computer after discarding that part of code? Because there are so many files involved.

**Comment by namannimmo10** [comment]
I don't know if this is a valid question or not but after every commit, I need to run the tests using `sympy.test("_sympy")` to check the errors.

**Comment by ShubhamKJha** [comment]
I would suggest to remove all the unnecessary code in compatibility. 
E.g, `integer_types` is not required (and I don't think it would be in future). In Python3 only `int` type is there. 
* You can easily `git grep integer_types` and look where it is used and replace it with `int`.
* Since `integer_types` is a tuple, the code would be like `type(<something>) in integer_types` change it to `type(<something>) is int` or `isinstance(<something>, integer_types)` change it to `isinstance(<something>, int)` and so on.
* Remove its import as `from compatibility import integer_types`. from the files.
* When all the occurrences of `integer_types` is changed, simply remove `integer_types` in compatibility.
* Confirm with `git grep` that `integer_types` is removed from codebase.
* Then test with `sympy.test()`.

Similarly, for others `string_types => str`, `class_types => type`, `long => int`. 
Whenever you finish one portion, commit it, so that it can be reviewed and discussed there.
Use a decent IDE, I would suggest to use PyCharm.

**Comment by namannimmo10** [comment]
Awesome! I'll follow that.

**Pull request merged**